internal/service/resources/serve.go

```diff
@@ -151,6 +167,8 @@ func Serve(config *api.ResourceServerConfig) error {
 	select {
 	case err := <-serverErrors:
 		return fmt.Errorf("error starting server: %w", err)
```

Comment:
likewise, if the server is stopped, we need to stop the collector

Reply:
thanks